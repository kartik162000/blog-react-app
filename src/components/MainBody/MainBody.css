
.card_area {
    width: 70%;
    margin: 0 auto;
    display: flex;
    flex-wrap: wrap;
    padding-top: 15px;
    min-height: 79vh;
    gap: 10px 50px;
    justify-content: space-between;
}
