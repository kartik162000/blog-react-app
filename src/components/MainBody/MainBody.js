import React from "react";
import Card from "../Card/Card";
import { stockData } from "../../assets/data";

import "./MainBody.css";
function MainBody() {
  return (
    <div>
      <div className="card_area">
        {stockData.map((data) => {
          return (
            <Card
              date={data.date}
              readingTime={data.readingTime}
              title={data.title}
              description={data.description}
              claps={data.claps}
              liked={data.liked}
              image={data.image}
            />
          );
        })}
      </div>
    </div>
  );
}

export default MainBody;
