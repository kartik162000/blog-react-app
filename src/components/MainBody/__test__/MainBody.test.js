import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import MainBody from "../MainBody";

describe("MainBody Component rendring test", () => {
  it("should render when component is mounted", () => {
    const { container } = render(<MainBody />);
    expect(container).toMatchSnapshot();
  });
});
