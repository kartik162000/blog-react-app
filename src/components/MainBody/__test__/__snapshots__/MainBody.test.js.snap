// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBody Component rendring test should render when component is mounted 1`] = `
<div>
  <div>
    <div
      class="card_area"
    >
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="abstract.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              2nd Januray, 2018
            </div>
            <div
              class="card-reading-time"
            >
              2 mins
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              The future of abstract art and the culture ...
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                10
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="outdoor.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              31st Januray, 2018
            </div>
            <div
              class="card-reading-time"
            >
              1 min
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              How painting outdoors can transform your 
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                1
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-red.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="young-painters.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              20th March, 2018
            </div>
            <div
              class="card-reading-time"
            >
              2 mins
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              10 young painters you need to know
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                15
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="perfect-art.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              22nd June, 2018
            </div>
            <div
              class="card-reading-time"
            >
              5 mins
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              Tips for creating the perfect art
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                11
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="inspired.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              4th Feburary, 2019
            </div>
            <div
              class="card-reading-time"
            >
              9 mins
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              Stay inspired: top artists blog ...
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                45
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-red.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-main"
      >
        <div
          class="card-image"
        >
          <img
            alt="image"
            src="reason.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-caption"
          >
            <div
              class="card-date"
            >
              10th Januray, 2019
            </div>
            <div
              class="card-reading-time"
            >
              2 mins
            </div>
          </div>
          <div
            class="card-title"
          >
            <h4>
              5 reasons why everyone should try an...
            </h4>
          </div>
          <div
            class="card-description"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <hr
            id="cardLine"
          />
          <div
            class="card-footer"
          >
            <div
              class="card-clap"
            >
              <img
                alt="clap"
                src="clapping.svg"
              />
              <div
                class="card-clap-count"
              >
                11
              </div>
            </div>
            <div
              class="card-heart"
            >
              <img
                alt="heart"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
